### ISSUE TITLE
Landing Page Design

### ISSUE THREAD

**Issue by shiffman** [CodingTrain/Rainbow-Code#354]
The landing page should include some elements from current http://thecodingtrain.com and more:

* patreon link
* subscribe link
* coding train merchandise store (shirts / mugs)
* coding train amazon shop

Also, I don't know that the full logo needs to take up so much space might be nice for people to see links to the videos, code, etc. without having to scroll down?

**Comment by shiffman** [comment]
Also add to landing page:

* Next live stream
* Latest coding challenge

**Comment by Merijn-DH** [comment]
Maybe something extra that pops up when you're live? Could be a youtube player or just a 'live' button thats only there when you are streaming?

**Comment by shiffman** [comment]
We can borrow from the current http://thecodingtrain.com/ in terms of the "latest video" design.

**Comment by Link2Twenty** [comment]
There are 3 ways to go about this.

1. Make a `Rainbow-Code/_jekyll/includes/2-base/landing.html` file and include it on the first page.
2. Edit `Rainbow-Code/FrontPage.md` to include.
3. A mixture of 1 and 2.

I think approach 3 is best

Landing | FrontPage
--- | ---
patreon link| merch store
subscribe link | amazon shop
Next live stream | Latest coding challenge*

_*The coding challenges section already has this._

Something like this would be a good start. Though below a certain screen sizes things would have to be done differently, the current breakpoints can be used to make a different layout.

![landing](https://user-images.githubusercontent.com/3534427/34909381-a0118edc-f897-11e7-8e05-ea9626f830ee.png)

**Comment by nilsweb** [comment]
@shiffman What do you think about this layout/mockup?

![6becbe17-d2c4-45d8-894a-d39cac25e065](https://user-images.githubusercontent.com/12577464/35186154-ade86b70-fe0f-11e7-9d80-2b7ad82542bb.png)

**Comment by shiffman** [comment]
This is looking great @nilsweb thank you! Some comments:

* maybe the logo stuff can be incorporated into the navbar? (to conserve space?) maybe not necessary.
* Instead of having a bio or anything on this page I'd rather just add an "about".
* We should add a link to the [amazon shop](https://www.amazon.com/shop/thecodingtrain)

![home](https://user-images.githubusercontent.com/191758/35195258-d0241ed0-fe8e-11e7-8412-aa4ad0b09f12.png)

Would love for any visual designers to weigh in!  (If it's helpful I have all the logo stuff as vector files I'll share via slack.)

**Comment by nilsweb** [comment]
@shiffman Thanks for your feedback!

- I'll experiment with this idea (especially in case of readability)
- a separate about page can easily be done
- I totally forgot about the Amazon shop... 😬 Should this also go under the "Support" section?
- I also realized that there is a signup form for a newsletter on the [current website](http://thecodingtrain.com). Should I add that as well?
- is there anything else that needs to go on the landing page?

So far, the structure would be:
- navigation + small banner
- latest coding challenge/stream and next stream
- support section
- other links (Nature of Code, Learning Processing and shiffmann.net)

**Comment by shiffman** [comment]
All sounds great! I'm not using the newsletter at all these days so let's skip that form. Something else we might include is icons in the header or footer for quick links to youtube, twitter, and patreon?

**Comment by nilsweb** [comment]
Alright, I'll post a new design over the next couple of days.

**Comment by nilsweb** [comment]
After a lot of cutting and pasting image snippets, I came up with something that looks like this:

![image6272](https://user-images.githubusercontent.com/12577464/35298592-efb1f272-0082-11e8-8c8f-42d9fe0b34fb.png)

I'm not quite sure... Feedback greatly appreciated!

**Comment by shiffman** [comment]
Love this so so much! Here are some quick thoughts:

1. I think the Learn to Code explanation stuff can be moved to the about page?
2. For me the "subscribe on youtube", "support on patreon" and "visit the merch store" are the links people are looking for most often so while I like the idea of "follow" / "support" I wonder if we could fit a single row of 4 and just push the twitter and github to smaller icons in the header nav?
3. Ideally I'd always have a "next live stream" scheduled to appear and maybe that can go where the "Learn to Code" block is?

**Comment by nilsweb** [comment]
1. Sure, no problem!
2. In theory, this is possible but I'm not quite sure if it will "feel right" since everything else is currently in a three column layout
3. see 1

**Comment by nilsweb** [comment]
Almost done! Looks like this at the moment:

![The Coding Train - Landing Page](https://user-images.githubusercontent.com/12577464/35540293-7e4c9236-0555-11e8-9bad-eff6c85fe394.jpg)

I decided to leave the "Learn to Code" block since it might be good to have some information about what's going on on this page, especially for people finding The Coding Train for the first time. Moreover, it would look a little "empty" without any form of introduction.
The video shown at the top would be the next live stream (or the latest coding challenge, if there is no stream planned). The videos below show the latest coding challenge and stream.
I also removed the border from the support links, since it looks a little "crammed" otherwise (too much shadow going on).

Honestly, at this point, everything looks a little weird to me 😬. I don't know what I should tweak, change, add, etc. so any kind of feedback would be great!
